apps/builder/app/builder/features/seo/site-settings.tsx

```diff
@@ -0,0 +1,210 @@
+          name="Name"
+          autoFocus
+        />
+      </Grid>
+
+      <Separator />
+
+      <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+        <Label htmlFor={ids.favicon} sectionTitle>
+          Favicon
+        </Label>
+        <Grid flow="column" gap={3}>
+          <Image
+            width={72}
+            height={72}
+            className={imgStyle()}
+            src={favIconUrl}
+            loader={imageLoader}
+          />
+
+          <Grid gap={2}>
+            <Text color="subtle">
+              Upload a 32 x 32 px image to display in browser tabs.
+            </Text>
+            <ImageControl
+              assetId={props.value.faviconAssetId ?? ""}
+              onAssetIdChange={handleChange("faviconAssetId")}
+            >
+              <Button id={ids.favicon} css={{ justifySelf: "start" }}>
+                Upload
+              </Button>
+            </ImageControl>
+          </Grid>
+        </Grid>
+      </Grid>
+
+      <Separator />
+
+      <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+        <Label htmlFor={ids.code} sectionTitle>
+          Custom Code
+        </Label>
+        <Text color="subtle">
+          Custom code and scripts will be added at the end of the &lt;head&gt;
+          tag to every page across the published site.
+        </Text>
+        <TextArea
+          id={ids.code}
+          rows={5}
+          autoGrow
+          maxRows={10}
+          value={props.value.code ?? ""}
+          onChange={handleChange("code")}
+        />
+      </Grid>
+      <div />
+    </Grid>
+  );
+};
+
+const SiteSettingsView = () => {
+  const [value, setValue] = useState(
+    pagesStore.get()?.meta ?? {
+      siteName: "",
+      faviconAssetId: "",
+      code: "",
+    }
+  );
+
+  const open = useStore($isSiteSettigsOpen);
+
+  const handleSave = useEffectEvent(() => {
+    serverSyncStore.createTransaction([pagesStore], (pages) => {
+      if (pages === undefined) {
+        return;
+      }
+
+      pages.meta = value;
+    });
+  });
+
+  return (
+    <Dialog
+      open={open}
+      onOpenChange={(isOpen) => {
+        handleSave();
+        $isSiteSettigsOpen.set(isOpen);
+      }}
+    >
+      <DialogContent
+        onBlur={handleSave}
+        // Left Aside panels (e.g., Pages, Components) use zIndex: theme.zIndices[1].
+        // For a dialog to appear above these panels, both overlay and content should also have zIndex: theme.zIndices[1].
+        css={{
+          zIndex: theme.zIndices[1],
+        }}
+        overlayCss={{
+          zIndex: theme.zIndices[1],
+        }}
+      >
+        <SiteSettingsContent value={value} onChange={setValue} />
+        {/* Title is at the end intentionally,
+         * to make the close button last in the tab order
+         */}
+        <DialogTitle>Site Settings</DialogTitle>
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+export const SiteSettings = () => {
```

Comment:
We normally call this *Container so that the SiteSettingsView can be just SiteSettings
Container would only have some state/data logic but no rendering

Reply:
Its the only exported class, and should have name related to its usage. When I use any imported component in React I dont mind is it just data class or container. Containers etc are good for internal implementations, but exposing this information out of component scope does give nothing